fix(codegen): missing parens for `in` in `for in` loop init

Dunqing opened the pull request:
Well, this looks like an accidental change before, `Context::empty().and_forbid_in(false)` does nothing at all.

Dunqing commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7705?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#7705** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7705?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7705?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#7703** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7703?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Boshen commented:
### Merge activity

* **Dec 6, 10:51 AM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Dunqing committed:
fix(codegen): missing parens for `in` in `for in` loop init (#7705)

Well, this looks like an accidental change before, `Context::empty().and_forbid_in(false)` does nothing at all.
